src/state/workload.rs

```diff
@@ -685,13 +708,34 @@ impl WorkloadStore {
         self.by_addr.get(addr).cloned()
     }
 
+    /// Finds the workload by workload information, as an arc.
+    pub fn find_by_info(&self, wl: &WorkloadInfo) -> Option<Arc<Workload>> {
+        // We do not have an index directly on the full workload info, but we can narrow it down
+        // to only workloads on the same node with the same identity -- a tiny set to iterate over
+        self.node_local_by_identity
+            .get(&WorkloadIdentity {
+                namespace: strng::new(&wl.namespace),
+                service_account: strng::new(&wl.service_account),
```

Comment:
I'm still not 100% sure what use `WorkloadIdentity` is here, it's only used in this file, it's a strict subset of `WorkloadInfo`, and it's _not_ an identity, as it could apply to N workloads, it's more of a query filter.

Either remove it, or include `wl.name` in `WorkloadIdentity` so `get` can filter down to a unique workload in all cases and we don't need the `filter` clause below?

Reply:
It is for what it says on the tin:
```
/// WorkloadIdentity provides information about a workloads identity. This is used in place of Identity
/// in places where we **do not have the full identity (no trust domain)** and know we are working with workloads specifically.
```

We don't have TD so we cannot use `Identity`. We want a mapping of `ThingThatIsAKeyForCertificate -> Workloads` (which is what `node_local_by_identity` is for), so if we add `name` to it we are no longer doing that.

The issue here really stems from the fact there are two things using node_local_by_identity. One is find_by_info which would prefer to have a name, and one is `was_last_identity_on_node` which **cannot** have `name`. We could make two indexes, but IMO its simpler and more efficient to have `find_by_info` do the filtering (its a cheap filter, there is at most 100 entries to filter and likely fewer)

Reply:
`and one is was_last_identity_on_node which cannot have name`

ugh. I missed `was_last_identity_on_node`. You're right, carry on.